homeassistant/components/matter/entity.py

```diff
@@ -55,24 +57,25 @@ def __init__(
         self._node_device = node_device
         self._device_type_instance = device_type_instance
         self.entity_description = entity_description
-        node = device_type_instance.node
         self._unsubscribes: list[Callable] = []
         # for fast lookups we create a mapping to the attribute paths
         self._attributes_map: dict[type, str] = {}
-        server_info = matter_client.server_info
+        server_info = cast(ServerInfo, self.matter_client.server_info)
         # The server info is set when the client connects to the server.
         assert server_info is not None
+        # create unique_id based on "Operational Instance Name" and endpoint/device type
         self._attr_unique_id = (
-            f"{server_info.compressed_fabric_id}-"
-            f"{node.unique_id}-"
+            f"{get_operational_instance_id(server_info, self._node_device.node())}-"
             f"{device_type_instance.endpoint}-"
             f"{device_type_instance.device_type.device_type}"
         )
 
     @property
     def device_info(self) -> DeviceInfo | None:
```

Comment:
Can we move this to init method and use `_attr_device_info` or do we need to recalculate this after entity addition?